### PR TITLE
Fix E2E tests

### PR DESCRIPTION
**Pull request by ldevulder**
- ci: bump k8s and Rancher Manager versions
- ci: fix action versions used

### CODE REVIEW

**Review by frelon — APPROVED** [review]
Very nice!

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -81,58 +81,56 @@ jobs:
           CHART=$(basename $FILE)
           echo "chart_name=$CHART" >> $GITHUB_OUTPUT
       - name: Upload chart
-        uses: actions/upload-artifact@v3.1.0
+        uses: actions/upload-artifact@v3
```
Comment (kkaempf) [inline .github/workflows/e2e.yaml:84]: This one too 🤔

File: Makefile [hunk 2]
```diff
@@ -146,12 +146,16 @@ validate:
...
+	kubectl cluster-info --context kind-$(CLUSTER_NAME)
+	kubectl label nodes --all --overwrite ingress-ready=true
+	kubectl label nodes --all --overwrite node-role.kubernetes.io/master=
+	kubectl get nodes -o wide
+	export EXTERNAL_IP=$$(kubectl get nodes -o jsonpath='{.items[].status.addresses[?(@.type == "InternalIP")].address}') && \
 	export BRIDGE_IP="172.18.0.1" && \
 	export CONFIG_PATH=$(E2E_CONF_FILE) && \
 	cd $(ROOT_DIR)/tests && $(GINKGO) -r -v ./e2e
-
+
```
Comment (kkaempf) [inline Makefile:158]: oops

**Review by frelon — APPROVED** [review]

File: .github/workflows/e2e.yaml (hunk 1 above)
Comment (ldevulder) [reply .github/workflows/e2e.yaml:84]: Same as for `cache@v3`.

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -14,7 +14,7 @@ jobs:
         uses: actions/setup-go@v3
         with:
           go-version-file: go.mod
-      - uses: actions/cache@v3.0.11
+      - uses: actions/cache@v3
```
[the remark this answers is not in the document]
Comment (ldevulder) [reply .github/workflows/build.yaml:17]: No, using `@v3` means always using the lastest version in the major one, so in that case v3.3.2.

File: Makefile (hunk 2 above)
Comment (ldevulder) [reply Makefile]: Thanks, fixed!

**Pull request merged**